scipy/special/cephes/incbet.c

```diff
@@ -72,15 +72,21 @@ static double incbcf(double a, double b, double x);
 static double incbd(double a, double b, double x);
 static double pseries(double a, double b, double x);
 
-double incbet(aa, bb, xx)
-double aa, bb, xx;
+double incbet(double aa, double bb, double xx)
 {
     double a, b, t, x, xc, w, y;
     int flag;
 
-    if (aa <= 0.0 || bb <= 0.0)
+    if (aa < 0.0 || bb < 0.0)
 	goto domerr;
 
+    if (aa == 0.0 && bb == 0.0)
+	    return (NPY_NAN);
+
+    if ((aa == 0.0 && bb > 0.0 && npy_isfinite(bb)) ||
+		    (aa > 0.0 && bb == 0.0 && npy_isfinite(aa)))
```

Comment:
Most of this looks right. I think this is the only problem.

```python3
from scipy import special

# To avoid mistakes when entering into Wolfram Alpha,
# make the argument orders match.
def BetaRegularized(x, a, b):
    return special.betainc(a, b, x)

# Values vere close to zero reported as 0.0
# Wolfram Alpha "(undefined)" reported as NaN
# When evaluating limits, a=0 or b=0 are replaced with x,
# and the limit is taken from the right. "-" means did not evaluate.

O = 1e-99
oo = 1e99
                                    # SciPy | Evaluated | Limit
print(BetaRegularized(O, O, O))     # 0.5   | NaN       | -
print(BetaRegularized(0.5, O, O))   # 0.5   | NaN       | -
print(BetaRegularized(1, O, O))     # 1.0   | NaN       | -
print(BetaRegularized(O, O, 1))     # 1.0   | 1         | 1 (both from the right)
print(BetaRegularized(0.5, O, 1))   # 1.0   | 1         | 1 (from the right)
print(BetaRegularized(1, O, 1))     # 1.0   | 1         | 1 (from the right)
print(BetaRegularized(O, 1, O))     # 0.0   | 0         | 0
print(BetaRegularized(0.5, 1, O))   # 0.0   | 0         | 0
print(BetaRegularized(1, 1, O))     # 1.0   | 0         | 1 (from the right)
```

So it looks like when `a > 0`, `b = 0`, and `x < 1`, the result should be 0. In this case, if `x = 1`, there is an argument to be made for making the output 1.0, but Wolfram Alpha gives a different result for the limit versus just evaluating at `b = 0`, and it gives `ComplexInfinity` in Mathematica itself, so it's not super clear.

Reply:
Thanks @mdhaber I think the stalling point for me here was mostly lack of concensus on the correct limit values. Given the ambiguity I'm not sure what's the best way to proceed for the edge cases.

Do you have thoughts or recommendations on how to resolve the ambiguity? 

I can make an arbitrary decision if that's what is required here but would prefer not to do so.

Reply:
ISTM that the only case (that's in scope) that is not clear-cut is `a > 0`, `b == 0`, and `x == 1`. For that, I think it would be fine to continue to return NaN. If someone enjoys this sort of thing, they can write up a paper about it; if we're convinced, we can change the value then. And at some point, there will be a meta-issue about auditing edge cases like this throughout `special`; this resolves the reported issue, so we can close gh-8411 once this merges.

So if you can update this to make the changes in `incbet.h` (and maybe make the changes for `betaincc`, too), I think it would be safe enough for me to merge it.